### PR TITLE
Use GitHubCommitStatusSetter

### PR DESCRIPTION
**Pull request by definite**
JIRA issue URL:  https://zanata.atlassian.net/browse/ZNTA-2035

Use GitHubCommitStatusSetter to report the status.

## Reviewer tasks

The following is based on the
[Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines).
As a reviewer, you should check the guidelines regularly for updates, and just
to refresh your memory.

The reviewer can use this list for reference to make sure
they have considered all the important points.

Use the checkboxes or not, whatever works best for you.

 - Code quality
   - [ ] Code passes style check (checkstyle/eslint)
   - [ ] Code is clear and concise
   - [ ] UI code is internationalised
   - [ ] Code has adequate comments where appropriate
   - Code is in an appropriate place
     - [ ] Classes are in appropriate packages
     - [ ] Code fits with class's responsibilities (SRP)
   - [ ] Configuration files are documented enough
   - If code is removed
     - [ ] No remaining code references the removed code
       - [ ] No orphaned rewrite rules
       - [ ] No orphaned config settings
     - [ ] No remaining comments refer to the removed code
     - [ ] Unused imports and libraries are removed
 - Testing
   - [ ] New code has tests
   - [ ] Changed code has tests
   - [ ] All tests pass
   - [ ] Test coverage stable or increased
 - Documentation
   - [ ] New features are documented
   - [ ] Docs removed for deleted features
   - [ ] Docs updated for all changed features
   - [ ] Developer/sysadmin docs updated if appropriate
 - If there are new dependencies
   - [ ] Does each new dependency pass all the
         [Considerations for new dependencies](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines#new-technologies-and-dependencies)?

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/450)

### CODE REVIEW

**Comment by aeng** [comment]
@definite where can we see the build result by this plugin in Pull request page?

**Comment by definite** [comment]
@aeng Find the little tick "✓" or "X" in front of the commit ID. Click them to get the result.

I think I still need to prevent the update before the build actually finished.

File: Jenkinsfile [hunk 1]
```diff
@@ -427,12 +430,14 @@ void integrationTests(String appserver) {
          */
 
         if (mvnResult != 0) {
-          currentBuild.result = 'UNSTABLE'
```
Comment (seanf) [inline Jenkinsfile]: I think we should keep this responsibility here, not in the Notifier.

File: Jenkinsfile [hunk 2]
```diff
@@ -345,13 +349,12 @@ timestamps {
         // run integration test tasks in parallel
         parallel tasks
 
-        // if the build is *still* green after running integration tests:
-        if (currentBuild.result == null) {
-          echo 'marking build as successful'
-          currentBuild.result = 'SUCCESS'
```
Comment (seanf) [inline Jenkinsfile]: I think we should keep this here.

File: Jenkinsfile [hunk 3]
```diff
@@ -444,10 +449,9 @@ void integrationTests(String appserver) {
           // Reduce workspace size
           sh "git clean -fdx"
         } else {
-          currentBuild.result = 'FAILED'
```
Comment (seanf) [inline Jenkinsfile]: We should keep this here.

**Comment by seanf** [comment]
Reviewed 1 of 2 files at r4, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Jenkinsfile, line 10 at r6](https://reviewable.io:443/reviews/zanata/zanata-platform/450#-Kqf2pRWIxaGz__SNLrB:-Kqf2pRWIxaGz__SNLrC:b-poasak) ([raw file](https://github.com/zanata/zanata-platform/blob/0a45ba1c93fa344fe9af0d35b8002d55f4b08430/Jenkinsfile#L10)):*
> ```Groovy
> // ansicolor(), Notifier, PullRequests, Strings
> // TODO: Change Back to master after zanata-pipeline-library merge
> @Library('zanata-pipeline-library@ZNTA-2035-GitHubReporting')
> ```

I'll just put a Reviewable discussion here so we don't forget.

---

*[Jenkinsfile, line 455 at r6](https://reviewable.io:443/reviews/zanata/zanata-platform/450#-Kqf3Z8-tp06mXnBh0_2:-Kqf3Z8-tp06mXnBh0_3:b-utgey6) ([raw file](https://github.com/zanata/zanata-platform/blob/0a45ba1c93fa344fe9af0d35b8002d55f4b08430/Jenkinsfile#L455)):*
> ```Groovy
>         } else {
>           notify.error("No integration test result for $appserver")
>           currentBuild.result = 'FAILURE'
> ```

Should this be FAILED or FAILURE?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/450)*

**Comment by seanf** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Jenkinsfile, line 315 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/450#-Kpsd1SmpW7aJKasZV7g-r4-315:-Kqf4Y1LnENz-EuowBvy:b-4327n9) ([raw file](https://github.com/zanata/zanata-platform/blob/34128ece657467f03b5fdab9d8033b84da0a959e/Jenkinsfile#L315)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

You may need to call notify.finish() here too, before aborting the pipeline. 

Or perhaps you should just call notify.failed() (and keep settting result = FAILURE) instead of calling notify.testResults? ie it might be better put back as it was.
</blockquote></details>

You should use the new method `notify.error` here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/450)*

**Comment by definite** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Jenkinsfile, line 455 at r6](https://reviewable.io:443/reviews/zanata/zanata-platform/450#-Kqf3Z8-tp06mXnBh0_2:-KqfXS-G4efbCHZkpiad:bm6ulw2) ([raw file](https://github.com/zanata/zanata-platform/blob/0a45ba1c93fa344fe9af0d35b8002d55f4b08430/Jenkinsfile#L455)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Should this be FAILED or FAILURE?
</blockquote></details>

Quote from the currentBuild in Global variable references
result
    typically SUCCESS, UNSTABLE, or FAILURE (may be null for an ongoing build)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/450)*

**Comment by seanf** [comment]
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/450)*

**Comment by seanf** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/450#-:-KqgJS53y1-lTXuTi4Lv:bnfp4nl)*

**Comment by spathare21** [comment]
Verified build pass looks fine

**Pull request merged**